src/utils/units.ts

```diff
@@ -118,20 +119,22 @@ const calculateGasUsedPercentage = (gasLimit: string, gasUsed: string) => {
   return gasUsedBN.div(gasLimitBN).multipliedBy(bigify(100));
 };
 
-const gasPriceToBase = (price: number) => toWei(price.toString(), getDecimalFromEtherUnit('gwei'));
+const gasPriceToBase = (price: string | number) =>
+  toWei(price.toString(), getDecimalFromEtherUnit('gwei'));
 
 const totalTxFeeToString = (gasPriceEther: string, gasLimit: string): string =>
-  parseFloat(fromWei(totalTxFeeToWei(gasPriceEther, gasLimit), 'ether')).toFixed(6);
+  bigify(fromWei(totalTxFeeToWei(gasPriceEther, gasLimit), 'ether')).toFixed(6);
 
 const totalTxFeeToWei = (gasPriceEther: string, gasLimit: string): Wei =>
-  new BN(parseInt(gasPriceEther, 10)).mul(new BN(parseInt(gasLimit, 10)));
+  new BN(gasPriceEther, 10).mul(new BN(gasLimit, 10));
 
 const gasStringsToMaxGasBN = (gasPriceGwei: string, gasLimit: string): BN => {
-  return gasPriceToBase(parseFloat(gasPriceGwei)).mul(new BN(gasLimit));
+  return gasPriceToBase(gasPriceGwei).mul(new BN(gasLimit));
 };
 
-const gasStringsToMaxGasNumber = (gasPriceGwei: string, gasLimit: string): number => {
-  return parseFloat(
+// @todo Rename
+const gasStringsToMaxGasNumber = (gasPriceGwei: string, gasLimit: string): BigNumber => {
```

Comment:
Can we enforce hex strings or BigNumbers here? Can we use `bigify` here?

Reply:
I think this is as clean as it can be with latest changes!